py_cui/__init__.py

```diff
@@ -158,36 +158,36 @@ def __init__(self, num_rows, num_cols, auto_focus_buttons=True,
 
         # Initialize grid, renderer, and widget dict
         self._grid                  = py_cui.grid.Grid(num_rows, num_cols, self._height, self._width, self._logger)
-        self._renderer              = None
-        self._border_characters     = None
-        self._stdscr                = None
-        self._widgets               = {}
+        self._stdscr: Any              = None
         self._refresh_timeout       = -1
+        self._border_characters: Optional[Dict[str,str]] = None
+        self._widgets: Dict[int,Optional['py_cui.widgets.Widget']] = {}
+        self._renderer: Optional['py_cui.renderer.Renderer'] = None
 
         # Variables for determining selected widget/focus mode
-        self._selected_widget       = None
+        self._selected_widget: Optional[int] = None
         self._in_focused_mode       = False
-        self._popup                 = None
+        self._popup: Any = None
         self._auto_focus_buttons    = auto_focus_buttons
 
         # CUI blocks when loading popup is open
         self._loading               = False
         self._stopped               = False
-        self._post_loading_callback = None
-        self._on_draw_update_func   = None
+        self._post_loading_callback: Optional[Callable[[],Any]] = None
+        self._on_draw_update_func: Optional[Callable[[],Any]]   = None
 
         # Top level keybindings. Exit key is 'q' by default
-        self._keybindings  = {}
```

Comment:
The values of this dict should be a Callable object I think

Reply:
I have changed this!